### PR TITLE
Basic Features

### PR DESCRIPTION
**Pull request by MayankMittal1**

### CODE REVIEW

**Comment by MusabShakeel576** [comment]
@MayankMittal1 which commands are you using to run examples files? It's throwing the following error while running `/example/address.js`:

```bash
fuse_wallet_web_sdk/node_modules/@ethersproject/bytes/lib/index.js:9
return !!(value.toHexString);
```

Btw a few small changes which could optimize this package/sdk:
- Create `.npmignore` file (or use `files` attribute in `package.json`) and add `/src` and `/example` folders into it to reduce the bundle size.
- Put `/dist` in `.gitignore`.

File: src/constants/variables.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+export class Variables {
+  static FUSE_CHAIN_ID = 122;
```
Comment (leonprou) [inline src/constants/variables.ts]: How can we run the SDK on Spark or staging if those are hard coded?

Comment (leonprou) [reply src/constants/variables.ts]: @SametSahin10 @NikolayRn
We have the same issue with a mobile SDK, right?

File: src/etherspot.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+import { ERC4337 } from 'userop/dist/constants';
+import { EOASignature, estimateUserOperationGas, getGasPrice } from 'userop/dist/preset/middleware';
+import {
+  EtherspotWalletFactory,
+  EtherspotWalletFactory__factory,
+  EtherspotWallet__factory,
+  EtherspotWallet as EtherspotWalletContract,
+} from '@etherspot/prime-sdk/dist/sdk/contracts';
+
+export const ETHERSPOT_FACTORY = '0x7f6d8F107fE8551160BD5351d5F1514A6aD5d40E';
```
Comment (leonprou) [inline src/etherspot.ts]: put to this in a config

File: src/modules/staking.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      }
+      return new Error('Failed to fetch encoded data for stake interaction');
+    } catch (e: any) {
+      return new Error(e);
+    }
+  }
+
+  public async unstake(unstakeRequestBody: IUnstakeRequestBody) {
+    try {
+      const response = await this._axios.post('/v0/staking/unstake', unstakeRequestBody);
```
Comment (leonprou) [inline src/modules/staking.ts]: There's no reason no to do this on the Dapp side

File: src/modules/staking.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+      }
+      return new Error('Error getting staking options');
+    } catch (e: any) {
+      return new Error(e);
+    }
+  }
+
+  public async stake(stakeRequestBody: IStakeRequestBody) {
+    try {
+      const response = await this._axios.post('/v0/staking/stake', stakeRequestBody);
```
Comment (leonprou) [inline src/modules/staking.ts]: There's no reason no to do this on the Dapp side

File: src/sdk.ts [hunk 5]
```diff
@@ -0,0 +1,628 @@
...
+        'Content-Type': 'application/json',
+      },
+      params: {
+        apiKey: publicApiKey,
+      },
+    });
+    this._initializeModules();
+  }
+
+  public static defaultTxOptions = new TxOptions('1000000', 10, false);
```
Comment (leonprou) [inline src/sdk.ts]: Can we explain those values?
I wonder if they should be part of the config

File: src/utils/contracts.ts [hunk 6]
```diff
@@ -0,0 +1,243 @@
...
+   * @param from is the address of the sender.
+   * @param to is the address of the recipient.
+   * @param data is the data to be sent with the transaction.
+   * @param nonce is the nonce of the transaction.
+   * @param value is the amount of Ether to be sent with the transaction.
+   * @param gasLimit is the gas limit of the transaction.
+   * @param gasPrice is the gas price of the transaction.
+   * @returns a hex string representing the signed transaction.
+   */
+  public static signOffChain(
```
Comment (leonprou) [inline src/utils/contracts.ts]: That's a method for the old SDK

File: src/utils/contracts.ts [hunk 7]
```diff
@@ -0,0 +1,243 @@
...
+import { bytesToHex, hexToBytes, keccak256 } from 'web3-utils';
+import { hexZeroPad, hexlify } from 'ethers/lib/utils';
+import { Variables } from '../constants/variables';
+
+/**
+ * A utility class providing methods to interact with contracts.
+ * `ContractsUtils` simplifies tasks such as reading from a contract,
+ * encoding data for contract calls, and signing off-chain transactions.
+ */
+export class ContractUtils {
```
Comment (leonprou) [inline src/utils/contracts.ts:9]: Hi @SametSahin10 
Are methods in this class are needed in the new SDK? Looks to me that some of them related to the old one

**Review by leonprou — CHANGES_REQUESTED** [review]
Looks good,
Please see my comments @MusabShakeel576 , @MayankMittal1 .
Also I want to loop in @SametSahin10 for the compability with the mobile version

File: src/constants/variables.ts (hunk 1 above)
Comment (SametSahin10) [reply src/constants/variables.ts]: This constant exists in the mobile SDK but it's not used anywhere. @leonprou

File: src/utils/contracts.ts (hunk 7 above)
Comment (SametSahin10) [reply src/utils/contracts.ts:9]: Some of them are used in the new SDK, yes. Some of the are only used by the old SDK. For example:
- `encodedDataForContractCall`
- `signOffChain`

@leonprou

File: src/constants/variables.ts (hunk 1 above)
Comment (MusabShakeel576) [reply src/constants/variables.ts]: @leonprou fixed, added env to Variables to make them soft coded in a new [Fix Basic Features PR](https://github.com/fuseio/fuse_wallet_web_sdk/pull/2).

File: src/etherspot.ts (hunk 2 above)
Comment (MusabShakeel576) [reply src/etherspot.ts]: @leonprou updated.

File: src/modules/staking.ts (hunk 3 above)
Comment (MusabShakeel576) [reply src/modules/staking.ts]: @leonprou fixed.

File: src/modules/staking.ts (hunk 4 above)
Comment (MusabShakeel576) [reply src/modules/staking.ts]: @leonprou fixed.

File: src/sdk.ts (hunk 5 above)
Comment (MusabShakeel576) [reply src/sdk.ts]: @leonprou fixed, moved them into Variables and remove TxOptions class.

File: src/utils/contracts.ts (hunk 6 above)
Comment (MusabShakeel576) [reply src/utils/contracts.ts]: @leonprou removed.

File: src/utils/contracts.ts (hunk 7 above)
Comment (leonprou) [reply src/utils/contracts.ts:9]: Thanks @SametSahin10 
@MusabShakeel576 please delete encodedDataForContractCall also

File: src/constants/variables.ts (hunk 1 above)
Comment (leonprou) [reply src/constants/variables.ts]: Thanks Musab

File: src/utils/contracts.ts (hunk 7 above)
Comment (MusabShakeel576) [reply src/utils/contracts.ts:9]: @leonprou deleted https://github.com/fuseio/fuse_wallet_web_sdk/pull/2/commits/9c1d9c64d43de502e7970987af569aff6972c641

**Review by MusabShakeel576 — COMMENTED** [review]
LGTM

**Review by leonprou — APPROVED** [review]

**Pull request merged**